### PR TITLE
For #6531 improve Verify Linking UI smoke tests coverage

### PR DESCRIPTION
**Pull request by AndiAJ**
For #6531 new `emailLinkTest` and `telephoneLinkTest` successfully passed 40x on Firebase

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt [hunk 1]
```diff
@@ -38,12 +44,14 @@ class WebControlsTest {
         webServer.start()
         featureSettingsHelper.setCfrForTrackingProtectionEnabled(false)
         featureSettingsHelper.setNumberOfTabsOpened(4)
+        Intents.init()
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt:47]: Is this needed? I know there's similar test on Fenix but I don't remember needing to initialize anything.

File: app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt [hunk 2]
```diff
@@ -38,12 +44,14 @@ class WebControlsTest {
...
     }
 
     @After
     fun tearDown() {
         webServer.shutdown()
         featureSettingsHelper.resetAllFeatureFlags()
+        Intents.release()
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt:54]: See above

File: app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt [hunk 3]
```diff
@@ -92,6 +100,38 @@ class WebControlsTest {
...
+
+    @SmokeTest
+    @Test
+    fun telephoneLinkTest() {
+        webServer.enqueue(createMockResponseFromAsset("htmlControls.html"))
+        val htmlControlsPage = webServer.url("htmlControls.html").toString()
+
+        searchScreen {
+        }.loadPage(htmlControlsPage) {
+            progressBar.waitUntilGone(waitingTime)
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt:127]: Is this needed for a local asset?

File: app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt [hunk 4]
```diff
@@ -92,6 +100,38 @@ class WebControlsTest {
...
+            clickLinkMatchingText("Telephone link")
+            progressBar.waitUntilGone(waitingTime)
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt:129]: Same here, on click, what are we waiting for here?

File: app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt [hunk 5]
```diff
@@ -415,6 +418,14 @@ object TestHelper {
         return customTabsIntent.intent
     }
 
+    fun assertNativeAppOpens(appPackageName: String, url: String) {
+        if (isPackageInstalled(appPackageName)) {
+            Intents.intended(IntentMatchers.toPackage(appPackageName))
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt:423]: `Intended` can throw `AssertionFailedError`, let's wrap in a `Try-Catch`

```Kotlin
try {
                intended(toPackage(appPackageName))
            } catch (e: AssertionFailedError) {
                e.printStackTrace()
            }
```

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt (hunk 3 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt]: Removed it, wanted to be sure that the page is fully loaded to be able to click the link.

File: app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt (hunk 5 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt]: Done

File: app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt (hunk 4 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt]: Removed it, got misled by the fact that after tapping the re-direct link the progress bar is loading.

File: app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt:52]: See above

File: app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/WebControlsTest.kt:45]: Without it the intent can't be verified.
`java.lang.NullPointerException: Intents not initialized. Did you forget to call init()?`

**Comment by AndiAJ** [comment]
Hi @AaronMT , made the changes, answered your questions and successfully re-ran both tests 15x on Firebase

**Pull request merged**